src/libs/sfdk/sdk.cpp

```diff
@@ -53,4 +90,59 @@ SdkPrivate::SdkPrivate(Sdk *q)
 
 SdkPrivate::~SdkPrivate() = default;
 
+Utils::FileName SdkPrivate::settingsFile(SettingsScope scope, const QString &basename)
+{
+    const QString prefix = scope == SessionScope
+        ? QString::fromLatin1(Constants::LIB_ID) + '-'
+        : QString();
```

Comment:
Even though this is syntactically valid, and probably does what you intend it to do, please don't write code like this. For beginner programmers like me, code like this is really difficult to read. There is really no need to put assignments and comparisons to a single line. ?: operator is difficult enough to parse as it is.

Reply:
You mean you would prefer

```
bool isSessionScope;
if (scope == SessionScope)
    isSessionScope = true;
else
    isSessionScope = false;

```
over

```
bool isSessionScope = scope == SessionScope;
```

or what?

Coding conventions exist to make code easier to read. When coding conventions are followed, you do not need to read every character of a code to understand its structure. This is the first knowledge a beginner programmer should adopt. Like breaking longer or otherwise complex lines with ternary operator involved, which is also a special case for the "one statement on a line" rule.

```
coreplugin/dialogs/addtovcsdialog.cpp-    QString addTo = files.size() == 1
coreplugin/dialogs/addtovcsdialog.cpp:            ? tr("Add the file to version control (%1)").arg(vcsDisplayName)
coreplugin/dialogs/addtovcsdialog.cpp-            : tr("Add the files to version control (%1)").arg(vcsDisplayName);
--
coreplugin/dialogs/externaltoolconfig.cpp-    const QString placeholderText = Utils::HostOsInfo::isWindowsHost()
coreplugin/dialogs/externaltoolconfig.cpp:            ? tr("PATH=C:\\dev\\bin;${PATH}")
coreplugin/dialogs/externaltoolconfig.cpp-            : tr("PATH=/opt/bin:${PATH}");
--
coreplugin/dialogs/saveitemsdialog.cpp-    const QDialogButtonBox::ButtonRole discardButtonRole = Utils::HostOsInfo::isMacHost()
coreplugin/dialogs/saveitemsdialog.cpp:            ? QDialogButtonBox::ResetRole : QDialogButtonBox::DestructiveRole;
coreplugin/dialogs/saveitemsdialog.cpp-
--
coreplugin/dialogs/shortcutsettings.cpp-    shortcutLabel->setToolTip(Utils::HostOsInfo::isMacHost()
coreplugin/dialogs/shortcutsettings.cpp:           ? QLatin1String("<html><body>")
coreplugin/dialogs/shortcutsettings.cpp-             + tr("Use \"Cmd\", \"Opt\", \"Ctrl\", and \"Shift\" for modifier keys. "
--
coreplugin/dialogs/shortcutsettings.cpp-    item->m_item->setForeground(2, hasCollision
coreplugin/dialogs/shortcutsettings.cpp:                                ? Utils::creatorTheme()->color(Utils::Theme::TextColorError)
coreplugin/dialogs/shortcutsettings.cpp-                                : commandList()->palette().window());
--
coreplugin/editormanager/openeditorswindow.cpp-    item->setIcon(0, !entry->fileName().isEmpty() && entry->document->isFileReadOnly()
coreplugin/editormanager/openeditorswindow.cpp:                  ? DocumentModel::lockedIcon() : m_emptyIcon);
coreplugin/editormanager/openeditorswindow.cpp-    item->setText(0, title);
--
coreplugin/editormanager/editormanager.cpp-        revertToSavedAction->setText(document->isModified()
coreplugin/editormanager/editormanager.cpp:                                     ? tr("Revert %1 to Saved").arg(quotedName)
coreplugin/editormanager/editormanager.cpp-                                     : tr("Reload %1").arg(quotedName));
--
coreplugin/editormanager/editormanager.cpp-    const QString windowTitleAddition = d->m_titleAdditionHandler
coreplugin/editormanager/editormanager.cpp:            ? d->m_titleAdditionHandler(filePath)
coreplugin/editormanager/editormanager.cpp-            : QString();
--
coreplugin/editormanager/editormanager.cpp-    const QString windowTitleVcsTopic = d->m_titleVcsTopicHandler
coreplugin/editormanager/editormanager.cpp:           ? d->m_titleVcsTopicHandler(filePath)
coreplugin/editormanager/editormanager.cpp-           : QString();
--
coreplugin/editormanager/editormanager.cpp-    const QString sessionTitle = d->m_sessionTitleHandler
coreplugin/editormanager/editormanager.cpp:           ? d->m_sessionTitleHandler(filePath)
coreplugin/editormanager/editormanager.cpp-           : QString();
--
coreplugin/editormanager/editormanager.cpp-    d->m_closeCurrentEditorContextAction->setText(entry
coreplugin/editormanager/editormanager.cpp:                                                    ? tr("Close \"%1\"").arg(quotedDisplayName)
coreplugin/editormanager/editormanager.cpp-                                                    : tr("Close Editor"));
coreplugin/editormanager/editormanager.cpp-    d->m_closeOtherDocumentsContextAction->setText(entry
coreplugin/editormanager/editormanager.cpp:                                                   ? tr("Close All Except \"%1\"").arg(quotedDisplayName)
coreplugin/editormanager/editormanager.cpp-                                                   : tr("Close Other Editors"));
--
coreplugin/find/basetextfind.cpp-            candidate.movePosition(options & QTextDocument::FindBackward
coreplugin/find/basetextfind.cpp:                                   ? QTextCursor::PreviousCharacter
coreplugin/find/basetextfind.cpp-                                   : QTextCursor::NextCharacter);
--
coreplugin/find/highlightscrollbarcontroller.cpp-    const double backgroundRatio = sizeDocInvisible
coreplugin/find/highlightscrollbarcontroller.cpp:            ? ((double)scrollBarBackgroundHeight / sizeDocInvisible) : 0;
coreplugin/find/highlightscrollbarcontroller.cpp-
--
coreplugin/find/highlightscrollbarcontroller.cpp-    const double handleRatio = sizeDocVisible
coreplugin/find/highlightscrollbarcontroller.cpp:            ? ((double)handleRect.height() / sizeDocVisible) : 0;
coreplugin/find/highlightscrollbarcontroller.cpp-
--
coreplugin/find/searchresulttreeitemdelegate.cpp-    QPalette::ColorGroup cg = option.state & QStyle::State_Enabled
coreplugin/find/searchresulttreeitemdelegate.cpp:                              ? QPalette::Normal : QPalette::Disabled;
coreplugin/find/searchresulttreeitemdelegate.cpp-    if (cg == QPalette::Normal && !(option.state & QStyle::State_Active))
--
coreplugin/find/searchresultwindow.cpp-    color.highlightBackground = highlightBackgroundColor.isValid()
coreplugin/find/searchresultwindow.cpp:            ? highlightBackgroundColor
coreplugin/find/searchresultwindow.cpp-            : textBackgroundColor;
coreplugin/find/searchresultwindow.cpp-    color.highlightForeground = highlightForegroundColor.isValid()
coreplugin/find/searchresultwindow.cpp:            ? highlightForegroundColor
coreplugin/find/searchresultwindow.cpp-            : textForegroundColor;
--
coreplugin/locator/locatorwidget.cpp-        const int highlightColumn = entry.highlightInfo.dataType == LocatorFilterEntry::HighlightInfo::DisplayName
coreplugin/locator/locatorwidget.cpp:                                                                 ? DisplayNameColumn
coreplugin/locator/locatorwidget.cpp-                                                                 : ExtraInfoColumn;
--
coreplugin/documentmanager.cpp-            fileDialogPath = defaultPath + (suggestedName.isEmpty()
coreplugin/documentmanager.cpp:                    ? QString()
coreplugin/documentmanager.cpp-                    : '/' + suggestedName);
--
coreplugin/editortoolbar.cpp-    d->m_editorList->setToolTip(document->filePath().isEmpty()
coreplugin/editortoolbar.cpp:                                ? document->displayName()
coreplugin/editortoolbar.cpp-                                : document->filePath().toUserOutput());
--
coreplugin/fancyactionbar.cpp-    const QIcon::Mode iconMode = isEnabled()
coreplugin/fancyactionbar.cpp:                                     ? ((isDown() || isChecked()) ? QIcon::Active : QIcon::Normal)
coreplugin/fancyactionbar.cpp-                                     : QIcon::Disabled;
--
coreplugin/fancyactionbar.cpp-                                      // state as they would do on the mode buttons.
coreplugin/fancyactionbar.cpp:                                      ? Theme::FancyTabWidgetEnabledUnselectedTextColor
coreplugin/fancyactionbar.cpp-                                      : Theme::FancyTabWidgetDisabledUnselectedTextColor));
--
coreplugin/iwizardfactory.cpp-            path = DocumentManager::useProjectsDirectory()
coreplugin/iwizardfactory.cpp:                       ? DocumentManager::projectsDirectory().toString()
coreplugin/iwizardfactory.cpp-                       : DocumentManager::fileDialogLastVisitedDirectory();
--
coreplugin/manhattanstyle.cpp-            const QColor color = creatorTheme()->color(enabled
coreplugin/manhattanstyle.cpp:                                                       ? Theme::MenuItemTextColorNormal
coreplugin/manhattanstyle.cpp-                                                       : Theme::MenuItemTextColorDisabled);
--
coreplugin/manhattanstyle.cpp-            pal.setBrush(QPalette::ButtonText, dis
coreplugin/manhattanstyle.cpp:                ? creatorTheme()->color(Theme::MenuBarItemTextColorDisabled)
coreplugin/manhattanstyle.cpp-                : creatorTheme()->color(Theme::MenuBarItemTextColorNormal));
--
coreplugin/manhattanstyle.cpp-                pal.setBrush(QPalette::Text, creatorTheme()->color(dis
coreplugin/manhattanstyle.cpp:                                                                   ? Theme::IconsDisabledColor
coreplugin/manhattanstyle.cpp-                                                                   : Theme::PanelTextColorLight));
--
coreplugin/manhattanstyle.cpp-                painter->setPen((option->state & State_Enabled)
coreplugin/manhattanstyle.cpp:                                  ? option->palette.color(QPalette::Foreground)
coreplugin/manhattanstyle.cpp-                                  : creatorTheme()->color(Theme::IconsDisabledColor));
--
coreplugin/manhattanstyle.cpp-                    const QColor hightLight = creatorTheme()->flag(Theme::FlatToolBars)
coreplugin/manhattanstyle.cpp:                            ? creatorTheme()->color(Theme::FancyToolBarSeparatorColor)
coreplugin/manhattanstyle.cpp-                            : StyleHelper::sidebarHighlight();
coreplugin/manhattanstyle.cpp-                    const QColor borderColor = drawLightColored
coreplugin/manhattanstyle.cpp:                            ? QColor(255, 255, 255, 180) : hightLight;
coreplugin/manhattanstyle.cpp-                    if (widget && widget->property("topBorder").toBool()) {
--
coreplugin/navigationwidget.cpp-    const char *trToolTip = d->m_side == Side::Left
coreplugin/navigationwidget.cpp:                                ? (isShown() ? Constants::TR_HIDE_LEFT_SIDEBAR : Constants::TR_SHOW_LEFT_SIDEBAR)
coreplugin/navigationwidget.cpp-                                : (isShown() ? Constants::TR_HIDE_RIGHT_SIDEBAR : Constants::TR_SHOW_RIGHT_SIDEBAR);
--
coreplugin/navigationwidget.cpp-    return side == Side::Left
coreplugin/navigationwidget.cpp:            ? Utils::Icons::CLOSE_SPLIT_LEFT.icon()
coreplugin/navigationwidget.cpp-            : Utils::Icons::CLOSE_SPLIT_RIGHT.icon();
--
coreplugin/outputpane.cpp-    int minimum = (d->m_splitter->orientation() == Qt::Vertical
coreplugin/outputpane.cpp:                   ? om->sizeHint().height() : om->sizeHint().width());
coreplugin/outputpane.cpp-    if (nonMaximizedSize() < minimum && !d->m_isMaximized)
--
coreplugin/outputpanemanager.cpp-        QRect r = creatorTheme()->flag(Theme::FlatToolBars)
coreplugin/outputpanemanager.cpp:                  ? rect() : rect().adjusted(numberAreaWidth(), 1, -1, -1);
coreplugin/outputpanemanager.cpp-        p.fillRect(r, c);
--
coreplugin/sidebar.cpp-    d->m_widgets.at(0)->setCloseIcon(d->m_widgets.size() == 1
coreplugin/sidebar.cpp:                                     ? Utils::Icons::CLOSE_SPLIT_LEFT.icon()
coreplugin/sidebar.cpp-                                     : Utils::Icons::CLOSE_SPLIT_TOP.icon());
--
coreplugin/sidebar.cpp-            d->m_widgets.at(0)->setCloseIcon(d->m_widgets.size() == 1
coreplugin/sidebar.cpp:                                             ? Utils::Icons::CLOSE_SPLIT_LEFT.icon()
coreplugin/sidebar.cpp-                                             : Utils::Icons::CLOSE_SPLIT_TOP.icon());
--
coreplugin/vcsmanager.cpp-    return files.size() == 1
coreplugin/vcsmanager.cpp:        ? tr("Add the file\n%1\nto version control (%2)?")
coreplugin/vcsmanager.cpp-              .arg(files.front(), vc->displayName())
--
coreplugin/vcsmanager.cpp-    return files.size() == 1
coreplugin/vcsmanager.cpp:        ? tr("Could not add the file\n%1\nto version control (%2)\n")
coreplugin/vcsmanager.cpp-              .arg(files.front(), vc->displayName())
```

Reply:
> You mean you would prefer
> 
> ```
> bool isSessionScope;
> if (scope == SessionScope)
>     isSessionScope = true;
> else
>     isSessionScope = false;
> ```
> 
> over
> 
> ```
> bool isSessionScope = scope == SessionScope;
> ```
> 
> or what?

If I have to choose between those two, then yes, I would prefer the if - else version. But this I would like the most:

`bool isSessionScope = (scope == SessionScope);`

The use of parenthesis improves the code readability a *lot*. The reader does not have to wonder about operator precedence and he can think about more important issues.

And also separating the boolean initialization and the ternary operator like this:

```
bool isSessionScope = (scope == SessionScope);
QString prefix = isSessionScope ? foo : bar;
```

again makes it easier for the poor fellow who has to fix bugs in the code. I don't really like the ternary operator, it never makes the code easier to read but just the opposite. But if one really can't be bothered to write if/else blocks, at least the ternary operators could be used in a fashion which is easy to read and understand. Also, having the boolean in a separate variable makes it easier to debug the code.

Yes, I know there are plenty of examples in Qt Creator sources where the code is far from readable and understandable. It doesn't make it any better though. The guys behind Qt Creator are really smart and good programmers, but I wouldn't say that they are considerate towards fellow programmers who have to read their code. To me readable and easy to understand code is always better than code which uses neat tricks to perform stuff using the least amount of keystrokes. 

I'm also quite sure there isn't a Qt Creator specific coding convention which would require the programmer to stuff as much as possible to a single statement. I might be wrong about this though.

I hope you don't read this so that I would think that more code is always the better - quite the opposite is actually true.

Also, please don't take these comments as requirements for changing your current code. Instead I'm asking that you could perhaps consider the reader more when writing your future code.